### DISCUSSION TITLE
Limit concurrency with webhook requests

### DISCUSSION THREAD

**Discussion by gstark** [gotenberg/gotenberg#539] [Q&A]
I'm trying to figure out how to limit/queue requests for `/forms/chromium/convert/url` when making webhook requests. Since the webhook returns immediately, I can't see a way to use a fronting HTTP reverse proxy to rate limit.

Is there a way to do this? I'm willing to work on a PR to pool/queue if there is a recommended approach.

**Comment by gulien** [comment]
Hello @gstark,

IMO the best option is to handle this outside of Gotenberg. For instance, a message queue on which you can handle throughput.

**Reply by gstark** [reply]
Yes, thank you, we are likely to do so.

**Reply by gstark** [reply]
I did submit a PR, #547, since when I use webhooks that send to AWS S3, my message queue system would have to detect when the S3 URL is posted.
